src/notation/view/widgets/editstyle.cpp

```diff
@@ -1473,6 +1474,59 @@ QString EditStyle::pageCodeForElement(const EngravingItem* element)
     }
 }
 
+QString EditStyle::subPageCodeForElement(const EngravingItem* element)
+{
+    IF_ASSERT_FAILED(element) {
+        return QString();
+    }
+
+    switch (element->type()) {
+    case ElementType::TEXT: {
```

Comment:
I think we don't need this outer switch statement here; instead, we can just check if the element is any of the text-related classes, by checking `element->isTextBase()`:
```
    if (element->isTextBase()) {
        switch (toTextBase(element)->textStyleType()) {
        case …:
```
Also, I think the inner switch should handle all possible text styles explicitly, i.e. no `default:` case. Almost all of them have a corresponding subpage.

Reply:
Ok so I have added almost all of the subpage codes that I could find from https://github.com/musescore/MuseScore/blob/d1724df9d82f053bf35ed1df2decdd678b2ac3da/src/notation/view/widgets/editstyle.cpp#L99C1-L99C1, but I couldn't find for these ones: 
* Gradual tempo change
* String tunings
* Harp pedal diagram
* Harp pedal text diagram

Reply:
Also I get this warning message when building as these types are not getting handled in switch for that we might need default?

```
In file included from /home/re1san/Dev/MuseScore/builds/Linux-Qt5.15.2-gcc_64-Make-RelWithDebInfo/src/notation/CMakeFiles/notation.dir/Unity/unity_4_cxx.cxx:34:
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp: In static member function ‘static QString mu::notation::EditStyle::subPageCodeForElement(const mu::notation::EngravingItem*)’:
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘DEFAULT’ not handled in switch [-Wswitch]
 1484 |         switch (toTextBase(element)->textStyleType()) {
      |                ^
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘TEMPO_CHANGE’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘DYNAMICS’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘STRING_TUNINGS’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘HARP_PEDAL_DIAGRAM’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘HARP_PEDAL_TEXT_DIAGRAM’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘TEXT_TYPES’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1484:16: warning: enumeration value ‘IGNORED_TYPES’ not handled in switch [-Wswitch]
/home/re1san/Dev/MuseScore/src/notation/view/widgets/editstyle.cpp:1651:1: warning: control reaches end of non-void function [-Wreturn-type]
 1651 | }
      | ^
```

Out of these I couldn't find subpage codes for `TEMPO_CHANGE, STRING_TUNINGS, HARP_PEDAL_DIAGRAM, HARP_PEDAL_TEXT_DIAGRAM`
as mentioned above. (I found for DYNAMICS but there is no such option in right click style menu)

Reply:
I looked a bit more closely, and it turns out that `allSubPageCodes` is not in sync with the actual list of subpages. The actual list is namely generated based on `editableTextStyles()` (see the constructor of EditStyle, around line 944).

So, the solution would be to find out what's in `editableTextStyles()`, and then update `allSubPageCodes` accordingly. Then update the switch in `subPageCodeForElement`.

There are a few TextStyleTypes that have indeed no subpage; I'd list those at the bottom of the switch and return `QString()` for them.
```
        switch (toTextBase(element)->textStyleType()) {
        case TextStyleType::TITLE:
            return "title";
        …
        case TextStyleType::DYNAMICS:
        case TextStyleType::DEFAULT:
        case TextStyleType::TEXT_TYPES:
        case TextStyleType::IGNORED_TYPES:
            return QString();
```

Reply:
Done